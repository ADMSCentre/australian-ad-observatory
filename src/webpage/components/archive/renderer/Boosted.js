import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faRocket } from '@fortawesome/free-solid-svg-icons';

const Boosted = ({ boosted }) =>
  boosted ? (
    <div className="btn btn-outline-success">
      <FontAwesomeIcon icon={faRocket} />
      Post was boosted
    </div>
  ) : null;

Boosted.displayName = 'Boosted';

Boosted.propTypes = { boosted: PropTypes.bool };

Boosted.defaultProps = { boosted: false };

export default Boosted;
