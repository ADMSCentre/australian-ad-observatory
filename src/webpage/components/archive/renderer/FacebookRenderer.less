.fb-item {
  padding: 0;
  width: 500px;
  background-color: #fff;
  border-radius: 5px;
  border: 1px solid #ddd;
  margin-top: 25px;
  margin-bottom: 15px;
  margin-right: 15px;

  &:first-child {
    padding: 12px 12px 0 12px;
  }

  .hidden_elem {
    display: none;
  }

  [data-testid="post_message"] {
    padding: 10px 0;
  }

  & > img {
    max-width: 90%;
    border: 0;
    margin: 0;
    padding: 0;
  }

  video {
    display: none;
  }

  div.mtm div div div span div img {
    width: 498px;
    height: auto;
    margin-left: -12px;
    border-top: 1px solid #ccc;
    border-bottom: 1px solid #ccc;
  }

  // Hide divs that have been otherwise rendering as stray periods.
  ._6spk {
    display: none;
  }

  // Don't allow user to click buttons or links.
  // a,
  // button {
  //   // pointer-events: none;
  // }

  h5 {
    font-size: 14px; // Match size of source bar w/ Facebook's size
    margin-bottom: 0; // Bring timestamp closer to the source bar title
  }

  // Add "arrow" in place of "to" in source bar.
  ._gb8.img::after {
    content: ">";
    font-style: normal;
  }

  // Remove word "to" in source bar.
  ._gb8 u {
    display: none;
  }

  ._6m6 {
    padding-top: 2px;
    margin-top: 3px;
    display: block;
    float: left;
    padding-bottom: 10px;

    a {
      color: #1d2129;
      font-size: 16px;
      font-weight: 600;

      &:hover {
        text-decoration: none;
        color: #1d2129;
      }
    }
  }

  // Bring author names inline with profile pics.
  ._6a._6b {
    display: inline-block;
    height: 20px !important;

    a {
      color: #385898;
      font-size: 14px;
      font-weight: 600;
      font-family: Helvetica Neue, Helvetica, Arial, sans-serif;
      -webkit-font-smoothing: subpixel-antialiased;
    }
  }

  ._ohe ._6a {
    display: block;
    float: right;
    padding-left: 18px;
    height: 20px;
    vertical-align: middle;

    a {
      border: 1px solid rgb(190, 195, 201);
      background-color: rgb(238, 240, 242);
      color: #4b4f56;
      padding: 5px 8px;
      border-radius: 2px;

      &:hover {
        text-decoration: none;
      }
    }
  }

  ._6m7 {
    margin-top: 3px;
    color: #606770;
    font-size: 14px;
    line-height: 20px;
    word-break: break-word;
    display: none;
    overflow-x: hidden;
  }

  div[data-testid="post_message"] p {
    color: #1d2129;
    font-size: 14px;
    font-weight: 400;
    margin-bottom: 5px;
  }

  .accessible_elem {
    display: none;
  }

  ._4r_y {
    height: 2px;
  }

  // Hide inconsequential artifacts.
  ._52c6 {
    display: none;
  }

  div ._3x-2 {
    margin: 0;
    padding: 0;
  }

  // Hide buttons.
  button {
    display: none;
  }

  // Profile pic wrapper.
  ._5pb8._1yz2._8o._8s.lfloat._ohe,
  ._38vo {
    display: block;
    float: left;
    width: 32px;
    margin-right: 8px;
    border-radius: 16px;
    overflow: hidden;
  }

  ._38vo {
    max-height: 32px;
  }

  ._38vo img {
    max-height: 32px;
  }

  ._5x46,
  ._3dp ._29k {
    /* Profile name wrapper */
    margin-bottom: 6px;
  }

  ._3ekx {
    display: block;
  }

  ._6lz {
    padding-top: 10px;
    color: #606770;
    font-size: 12px;
  }

  .uiScaledImageContainer {
    max-width: 100%;
    width: auto;
    height: auto;
  }

  .scaledImageFitWidth {
    // max-width: 100%;
    // width: auto;
    // height: auto;
  }

  .ellipsis {
    line-height: 14px;
    text-transform: uppercase;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  .card-title {
    display: none;
  }

  .card-subtitle {
    display: none;
  }

  &.fb-publicUserPost {
    // background-color: #e1edec;
    div.mtm div div div span div div {
      display: none;
    }
  }

  &.fb-publicPagePost {
    // background-color: #e1edec;
    div.mtm div div div span div div {
      display: none;
    }
  }

  &.fb-sponsoredPost {
    // background-color: #f1edec;

    a[role="button"] {
      //border: 1px solid;
      border-radius: 2px;
      box-sizing: content-box;
      font-size: 12px;
      -webkit-font-smoothing: antialiased;
      font-weight: bold;
      justify-content: center;
      padding: 0 8px;
      position: relative;
      text-align: center;
      text-shadow: none;
      vertical-align: middle;
    }
  }

  ul.uiList {
    margin-left: -52px;
  }

  ul.uiList li {
    display: none;
  }

  ul.uiList li:first-child {
    display: inline-block;
    border-top: 1px solid #ccc;
    width: 100%;
  }

  ul.uiList li img {
    width: 498px;
    height: auto;
  }

  ._5eut,
  ._42gt {
    display: none;
  }

  ._4lpe,
  ._3htz,
  ._5mly,
  ._40ja,
  ._45oh {
    display: none;
  }
}

.targeting {
  margin-left: 5px;
  width: 500px;

  h4 {
    font-size: 12px;
    font-weight: 600;
    margin: 0;
    padding: 0;
  }

  .reason {
    margin-right: 10px;

    .reason-name {
      padding-left: 5px;
    }
  }
}
