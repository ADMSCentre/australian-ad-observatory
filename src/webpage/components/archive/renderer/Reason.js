import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faUsers,
  faGlobe,
  faTransgenderAlt,
  faMapMarker,
  faDotCircle
} from '@fortawesome/free-solid-svg-icons';

const REASONS = {
  WAISTUICustomAudienceType: {
    name: 'Custom Audience',
    icon: faUsers
  },
  WAISTUILocaleType: {
    name: 'Locale',
    icon: faGlobe
  },
  WAISTUIAgeGenderType: {
    name: 'Age and Gender',
    icon: faTransgenderAlt
  },
  WAISTUILocationType: {
    name: 'Location',
    icon: faMapMarker
  }
};

const Reason = ({ reason }) =>
  REASONS[reason] ? (
    <div className="reason badge badge-primary">
      <FontAwesomeIcon icon={REASONS[reason].icon} />
      <span className="reason-name">{REASONS[reason].name}</span>
    </div>
  ) : (
    <div className="reason badge badge-primary">
      <FontAwesomeIcon icon={faDotCircle} />
      <span className="reason-name">{reason}</span>
    </div>
  );

Reason.displayName = 'Reason';

Reason.propTypes = {
  reason: PropTypes.string.isRequired
};

export default Reason;
