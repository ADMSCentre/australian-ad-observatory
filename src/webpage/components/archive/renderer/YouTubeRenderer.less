.yt-item {
  &.yt-instreamVideoAd {
    background-color: #e1edec;
  }

  &.yt-videowallIframeCompanionAd {
    background-color: #f1edec;
  }

  &.yt-actionCompanionAd {
    background-color: #e2edec;
  }

  &.yt-imageCompanionAd {
    background-color: #f2edec;
  }

  &.yt-adActionInterstitial {
    background-color: #e3edec;
  }

  &.yt-invideoOverlayAd {
    background-color: #f3edec;
  }
}
