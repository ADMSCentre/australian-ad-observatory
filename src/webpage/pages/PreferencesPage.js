import React, { Fragment } from 'react';

import LocalePreferences from 'common/components/LocalePreferences';
import PermissionCheckbox from 'webpage/components/PermissionCheckbox';
import Deck from 'webpage/components/Deck';
import Card from 'webpage/components/Card';

import * as KEYS from 'common/keys';
import {
  GENERAL_PERMISSION_KEYS,
  FACEBOOK_PERMISSION_KEYS,
  YOUTUBE_PERMISSION_KEYS
} from 'common/constants';

const PERMISSIONS = JSON.parse(process.env.PERMISSIONS);

const makePref = (acc, elem) => {
  const key = KEYS[elem];
  const pref = PERMISSIONS[elem];
  if (key && pref) acc[key] = pref;
  return acc;
};

const GENERAL_PERMISSIONS = GENERAL_PERMISSION_KEYS.reduce(makePref, {});
const FACEBOOK_PERMISSIONS = FACEBOOK_PERMISSION_KEYS.reduce(makePref, {});
const YOUTUBE_PERMISSIONS = YOUTUBE_PERMISSION_KEYS.reduce(makePref, {});

const checkboxen = (perms, props) =>
  Object.keys(perms).map(perm => (
    <PermissionCheckbox
      key={perm}
      storageKey={perm}
      label={perms[perm].label}
      hintText={perms[perm].hintText}
      defaultValue={perms[perm].defaultValue}
      {...props}
    />
  ));

const CARDS = {
  locale: {
    title: 'Your Settings',
    subtitle:
      "Setting your language and location help us ensure we're handling your data responsibly and analyzing it correctly."
  },
  general: {
    title: 'General Preferences',
    subtitle: 'These settings apply to all data shared by the extension',
    permissions: GENERAL_PERMISSIONS
  },
  facebook: {
    title: 'Facebook Preferences',
    subtitle: 'These options control what data you share from your Facebook feed',
    permissions: FACEBOOK_PERMISSIONS
  },
  youtube: {
    title: 'YouTube Preferences',
    subtitle: 'These options control what data you share when you are using YouTube',
    permissions: YOUTUBE_PERMISSIONS
  }
};

const makeCard = (key, content) => (
  <Card title={CARDS[key].title} subtitle={CARDS[key].subtitle}>
    {content || checkboxen(CARDS[key].permissions)}
  </Card>
);

// no need for a ConsentAwareView as this page is not visible unless you
// have granted latest consent.
const PreferencesPage = () => (
  <Fragment>
    <h4 className="mt-4">Sharing Preferences</h4>
    <p>
      You have control over what you share with this project. To change what data you wish to share,
      check or uncheck the box to enable or disable sharing that type of data.
    </p>
    <div>
      <Deck>
        {makeCard('general')}
        {makeCard('locale', <LocalePreferences />)}
      </Deck>
    </div>
    <div className="mt-4">
      <Deck>
        {makeCard('facebook')}
        {makeCard('youtube')}
      </Deck>
    </div>
    <footer
      className="mt-4"
      dangerouslySetInnerHTML={{ __html: process.env.PREFERENCES_PAGE_FOOTER }}
    />
  </Fragment>
);
PreferencesPage.displayName = 'PreferencesPage';

export default PreferencesPage;
