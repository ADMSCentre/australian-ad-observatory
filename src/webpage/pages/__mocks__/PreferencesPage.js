import React from 'react';

const PreferencesPage = () => <h1>Preferences Page goes here</h1>;

export default PreferencesPage;
