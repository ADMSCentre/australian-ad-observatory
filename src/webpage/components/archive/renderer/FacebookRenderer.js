import React from 'react';
import PropTypes from 'prop-types';

import Reason from './Reason';
import Boosted from './Boosted';

import './FacebookRenderer.less';

const itemShape = {
  id: PropTypes.string.isRequired,
  itemId: PropTypes.string.isRequired,
  itemType: PropTypes.string.isRequired,
  payload: PropTypes.shape({
    contentHtml: PropTypes.string.isRequired
  }).isRequired
};

const AdTargeting = ({ item }) => {
  if (!item.payload.adTargetingData) return null;

  const {
    data: { waist_targeting_data: targetingData, waist_is_marketplace_boosted_listing: boosted }
  } = item.payload.adTargetingData;
  const targetingReasons = targetingData.map(({ __typename }) => __typename);

  return (
    <div className="targeting">
      <h4>Ad Targeting</h4>
      <Boosted boosted={boosted} />
      {targetingReasons.map(reason => (
        <Reason key={reason} reason={reason} />
      ))}
    </div>
  );
};
AdTargeting.displayName = 'Ad targeting';

AdTargeting.propTypes = {
  item: PropTypes.shape(itemShape).isRequired
};

const FacebookRenderer = ({ item }) => {
  const {
    id,
    itemId: _itemId,
    itemType,
    payload: { contentHtml: html }
  } = item;

  const classes = `ati-item fb-item fb-${itemType}`;

  return (
    <div className="archive-item">
      <div id={id} className={classes} dangerouslySetInnerHTML={{ __html: html }} />
      <AdTargeting item={item} />
    </div>
  );
};

FacebookRenderer.propTypes = {
  item: PropTypes.shape(itemShape).isRequired
};

export default FacebookRenderer;
