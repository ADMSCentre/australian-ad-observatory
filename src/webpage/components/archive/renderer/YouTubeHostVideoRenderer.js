import React from 'react';
import PropTypes from 'prop-types';
import Player from 'youtube-embed-video';

import Card from '../../Card';

import './YouTubeRenderer.less';

const YouTubeHostVideoRenderer = ({ item }) => {
  const classes = 'ati-item yt-item';

  return (
    <Card title={item.title} subtitle={item.author}>
      <div className={classes}>
        <Player videoId={item.id} suggestions={false} enhancedPrivacy={true} size="small" />
      </div>
    </Card>
  );
};

YouTubeHostVideoRenderer.displayName = 'YouTubeHostVideoRenderer';

YouTubeHostVideoRenderer.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    channelId: PropTypes.string,
    url: PropTypes.string,
    author: PropTypes.string,
    title: PropTypes.string
  }).isRequired // determine a precise shape for this.
};

export default YouTubeHostVideoRenderer;
