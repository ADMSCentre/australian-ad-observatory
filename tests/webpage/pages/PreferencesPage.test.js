import React from 'react';
import renderer from 'react-test-renderer';
import PreferencesPage from 'webpage/pages/PreferencesPage';

jest.mock('webpage/components/PermissionCheckbox');
jest.mock('common/components/LocalePreferences');

let tree;

beforeAll(() => {
  tree = renderer.create(<PreferencesPage />);
});

it('rendered correctly', () => {
  expect(tree).toMatchSnapshot();
});
