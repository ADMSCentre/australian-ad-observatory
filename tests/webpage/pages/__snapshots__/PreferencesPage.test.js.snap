// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered correctly 1`] = `
Array [
  <h4
    className="mt-4"
  >
    Sharing Preferences
  </h4>,
  <p>
    You have control over what you share with this project. To change what data you wish to share, check or uncheck the box to enable or disable sharing that type of data.
  </p>,
  <div>
    <div
      className="card-deck"
    >
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          General Preferences
        </div>
        <div
          className="card-body"
        >
          <div
            className="small mb-3"
          >
            These settings apply to all data shared by the extension
          </div>
          <div
            className="mt-3"
          >
            <input
              id="user_share_diagnostic_data"
              name="mocked-permission-checkbox"
              type="checkbox"
            />
            <input
              id="show_debug_data"
              name="mocked-permission-checkbox"
              type="checkbox"
            />
          </div>
        </div>
      </div>
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          Your Settings
        </div>
        <div
          className="card-body"
        >
          <div
            className="small mb-3"
          >
            Setting your language and location help us ensure we're handling your data responsibly and analyzing it correctly.
          </div>
          <div
            className="mt-3"
          >
            <div
              name="locale-preferences"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="mt-4"
  >
    <div
      className="card-deck"
    >
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          Facebook Preferences
        </div>
        <div
          className="card-body"
        >
          <div
            className="small mb-3"
          >
            These options control what data you share from your Facebook feed
          </div>
          <div
            className="mt-3"
          >
            <input
              id="facebook_share_sponsored_posts"
              name="mocked-permission-checkbox"
              type="checkbox"
            />
            <input
              id="facebook_show_collection_status"
              name="mocked-permission-checkbox"
              type="checkbox"
            />
          </div>
        </div>
      </div>
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          YouTube Preferences
        </div>
        <div
          className="card-body"
        >
          <div
            className="small mb-3"
          >
            These options control what data you share when you are using YouTube
          </div>
          <div
            className="mt-3"
          >
            <input
              id="youtube_share_watched_videos"
              name="mocked-permission-checkbox"
              type="checkbox"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="mt-4"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Footy!</p>",
      }
    }
  />,
]
`;
